no_std: Add support for deprecated LayoutErr

Been recently forging a small library, and decided to use this library but along the way got to see this:
```
error[E0412]: cannot find type `LayoutError` in module `core::alloc`
   --> /home/amin/.cache/cargo/registry/src/github.com-1ecc6299db9ec823/custom_error-1.9.1/src/error.rs:40:29
    |
40  | impl Error for core::alloc::LayoutError {}
    |                             ^^^^^^^^^^^ help: a struct with a similar name exists: `LayoutErr`
    |
   ::: /home/amin/.local/share/rustup/xtensa/rustc/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:399:1
    |
399 | pub struct LayoutErr {
    | -------------------- similarly named struct `LayoutErr` defined here

error: aborting due to previous error
```

The direct cause of this seems to be the fact that I currently am using a custom rustc, stuck on `1.48.0`, and [this deprecation](https://github.com/rust-lang/rust/blob/master/library/core/src/alloc/layout.rs#L401). Would really appreciate your consideration if possible to add support for this, thanks.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -23,6 +23,8 @@ jobs:
     - uses: actions-rs/toolchain@v1
       with:
         toolchain: ${{ matrix.rust }}
+        default: true
+        override: true
     
     - name: test
       uses: actions-rs/cargo@v1

--- a/src/error.rs
+++ b/src/error.rs
@@ -37,7 +37,7 @@ pub trait Error: Debug + Display {
     }
 }
 
-impl Error for core::alloc::LayoutError {}
+impl Error for core::alloc::LayoutErr {}
 impl Error for core::array::TryFromSliceError {}
 impl Error for core::cell::BorrowError {}
 impl Error for core::cell::BorrowMutError {}

